### DISCUSSION TITLE
How to install on Rails 7?

### DISCUSSION THREAD

**Discussion by RyanJWolfe** [themesberg/flowbite#73] [Q&A]
How can I install Flowbite on a fresh Rails 7 app that is already configured with Tailwind?

**Comment by kkurcz** [comment]
Did you figure it out in the end?

**Comment by seanbjornsson** [comment]
Generally, you'd either
1) [Use webpacker](https://github.com/themesberg/flowbite/issues/88)
2) or [use the cdn](https://github.com/themesberg/flowbite#include-via-cdn)

However, I think both are currently broken in a new Rails 7 app with Turbo on. 
See this issue here :point_right: https://github.com/themesberg/flowbite/issues/88

**Comment by seanbjornsson** [comment]
Has anyone gotten this working using only importmaps? This is my goal but I haven't been able to achieve this yet. 
I think it would be possible if issue https://github.com/themesberg/flowbite/issues/88 wasn't a thing.

**Reply by seanbjornsson** [reply]
For now I'm just going to pull down the source file into my app/javascript/ folder, and return to this in a few months...

**Reply by reefdog** [reply]
@seanbjornsson Were you ever able to get this working? It's _exactly_ what I'm struggling with to use Flowbite in a Node-less, all-importmap Rails project.

The #88 blocker I can get around by vendoring the patched JS library (like you mentioned), even though I don't love having to manually update it. But the other part is extending the Tailwind config with `flowbite/plugin`. Have you been able to lick that yet, and how?

**Reply by seanbjornsson** [reply]
@reefdog , if I remember correctly, that was only to get the JS working. So it's an either-or situation. Because I manually added and patched the js file, I didn't need to add the plugin. 

But you are probably now closer to this than I am now. Is there additional functionality you're looking for in the plugin?

**Reply by reefdog** [reply]
@seanbjornsson Hrm, my understanding was the JS library was for providing the JS functionality, while the Tailwind plugin was for extending the base Tailwind styles with some additional classes and some overrides. If that's not the case, oof I complicated things!

**Reply by reefdog** [reply]
Ah yeah, so here's the source for the plugin: https://github.com/themesberg/flowbite/blob/main/plugin.js
